MailCore/API/MailApiFetcher/MailApiFetcher+Common.swift

```diff
@@ -139,6 +139,23 @@ public extension MailApiFetcher {
         try await perform(request: authenticatedRequest(.folders(uuid: mailbox.uuid), method: .post, parameters: folder))
     }
 
+    @discardableResult
+    func delete(mailbox: Mailbox, folder: Folder) async throws -> Empty? {
+        try await perform(request: authenticatedRequest(
+            .folder(uuid: mailbox.uuid, folderUUID: folder.remoteId),
+            method: .delete
+        ))
+    }
+
+    func modify(mailbox: Mailbox, folder: Folder, name: String) async throws -> Folder {
+        let newName = ModifyFolder(name: name)
+        return try await perform(request: authenticatedRequest(
+            .modifyFolder(mailboxUuid: mailbox.uuid, folderId: folder.remoteId),
+            method: .post,
+            parameters: newName
```

Comment:
You can use parameters directly instead of creating a struct
```suggestion
            parameters: ["name": name]
```